### ISSUE TITLE
False positive "use of moved value" when destructuring array of move-only types

### ISSUE THREAD

**Issue by oliver-giersch** [intellij-rust/intellij-rust#3949]
## Environment

* **Intellij-Rust plugin version:**  v.0.2.98.2125-191
* **Rust toolchain version:** 1.35.0
* **IDE name and version:** CLion #CL-191.7479.33
* **Operating system:** macOS Mojave 10.14.5

## Problem description

False positive when destructuring arrays of move-only types.

## Steps to reproduce
 
```rust
let box_arr = [Box::new('a'), Box::new('b'), Box::new('c')];
let [a, b, c] = box_arr; // falsely reports "use of moved value"
```
